### PR TITLE
NfDiv-2916 and NFDiv-2955

### PR DESCRIPTION
**Pull request by BWeeks-JusticeGovUK**
https://tools.hmcts.net/jira/browse/NFDIV-2916
https://tools.hmcts.net/jira/browse/NFDIV-2955 (Welsh Translation)

### CODE REVIEW

**Review by mapingo — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java [hunk 1]
```diff
@@ -55,9 +66,13 @@ public void generateAndUpdateCaseData(final CaseData caseData,
 
         log.info("Generating Judicial Separation Conditional Order Refused Cover Letter for case id {} ", caseId);
 
+        final DocumentType coverLetter = caseData.getApplication().isPaperCase() && applicant.isRepresented()
```
Comment (amoriartyMOJ) [inline src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java:69]: Is the isPaperCase() check required here? JS cases are always paper / offline

File: src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java [hunk 2]
```diff
@@ -75,22 +108,33 @@ private Map<String, Object> templateContent(final CaseData caseData, final Long
...
 
-        templateContent.put(FIRST_NAME, applicant.getFirstName());
-        templateContent.put(LAST_NAME, applicant.getLastName());
-        templateContent.put(ADDRESS, applicant.getAddress());
+        templateContent.put(FEEDBACK, conditionalOrderCommonContent.generateLegalAdvisorComments(caseData.getConditionalOrder()));
 
-        if (caseData.getDivorceOrDissolution().isDivorce()) {
-            templateContent.put(MARRIAGE_OR_CIVIL_PARTNERSHIP, MARRIAGE);
-            templateContent.put(DIVORCE_OR_CIVIL_PARTNERSHIP, DIVORCE);
+        if (caseData.getApplication().isPaperCase() && applicant.isRepresented()) {
```
Comment (amoriartyMOJ) [inline src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java:113]: Same again, isPaperCase check not required?

File: src/main/java/uk/gov/hmcts/divorce/legaladvisor/service/printer/AwaitingAmendedApplicationPrinter.java [hunk 3]
```diff
@@ -79,15 +87,22 @@ public void sendLetters(final CaseData caseData, final Long caseId, final Applic
         }
     }
 
-    private List<Letter> awaitingAmendedApplicationLetters(final CaseData caseData) {
+    private Boolean isPaperCaseAndApplicantRepresented(final CaseData caseData, final Applicant applicant) {
```
Comment (amoriartyMOJ) [inline src/main/java/uk/gov/hmcts/divorce/legaladvisor/service/printer/AwaitingAmendedApplicationPrinter.java:90]: Worth checking everywhere this is done - if we reach a printer class it's already confirmed offline / paper case

**Review by amoriartyMOJ — APPROVED** [review]
Approved, left 3 comments but realised it's mentioned in a lot of places - isPaperCase can be removed from this PR where it's used as all JS cases are offline, by using the isPaperCase function it makes it seem like it's possible for an alternative online path but thats not possible

File: src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java (hunk 1 above)
Comment (BWeeks-JusticeGovUK) [reply src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java]: Resolved

File: src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java (hunk 2 above)
Comment (BWeeks-JusticeGovUK) [reply src/main/java/uk/gov/hmcts/divorce/document/content/GenerateJudicialSeparationCORefusedForAmendmentCoverLetter.java]: Resolved

File: src/main/java/uk/gov/hmcts/divorce/legaladvisor/service/printer/AwaitingAmendedApplicationPrinter.java (hunk 3 above)
Comment (BWeeks-JusticeGovUK) [reply src/main/java/uk/gov/hmcts/divorce/legaladvisor/service/printer/AwaitingAmendedApplicationPrinter.java]: Good point - I was being overly defensive.  Resolved.

**Comment by BWeeks-JusticeGovUK** [comment]
> Approved, left 3 comments but realised it's mentioned in a lot of places - isPaperCase can be removed from this PR where it's used as all JS cases are offline, by using the isPaperCase function it makes it seem like it's possible for an alternative online path but thats not possible

I was being overly defensive here - I think I've found and resolved all of these now.

**Pull request merged**